packages/apollo-gateway/src/QueryPlan.ts

```diff
@@ -38,20 +38,63 @@ export interface ParallelNode {
 export interface FetchNode {
   kind: 'Fetch';
   serviceName: string;
-  selectionSet: SelectionSetNode;
-  variableUsages?: { [name: string]: VariableDefinitionNode };
-  requires?: SelectionSetNode;
-  internalFragments: Set<FragmentDefinitionNode>;
-  source: string;
+  variableUsages?: string[];
+  requires?: SelectionNode[];
+  operation: string;
 }
+
 export interface FlattenNode {
   kind: 'Flatten';
   path: ResponsePath;
   node: PlanNode;
 }
 
+export type SelectionNode = FieldNode | InlineFragmentNode;
+
+export interface FieldNode {
+  readonly kind: 'Field';
+  readonly alias?: string;
+  readonly name: string;
+  readonly selections?: SelectionNode[];
+}
+
+export interface InlineFragmentNode {
+  readonly kind: 'InlineFragment';
+  readonly typeCondition?: string;
+  readonly selections: SelectionNode[];
+}
+
 export function serializeQueryPlan(queryPlan: QueryPlan) {
   return prettyFormat(queryPlan, {
     plugins: [queryPlanSerializer, astSerializer],
   });
 }
+
+export function getResponseName(node: FieldNode): string {
+  return node.alias ? node.alias : node.name;
+}
+
+export const trimSelectionNodes = (selections: readonly GraphQLJSSelectionNode[]): SelectionNode[] => {
+  let remapped: SelectionNode[] = [];
+
+  // intentionally leaving out fragment spread from the input selections, because I don't
+  // THINK that's possible to see in the final query plan -- prove me wrong
```

Comment:
If it's not clear right now, we should probably write a test scenario to verify:

1. To maintain that behavior.
2. To point to in validating this. (and so we can remove any doubt / this comment).

Reply:
🤔  @JakeDawkins Can you double click on this (plausible) thread that @trevor-scheer is referring to?

Reply:
Going to mark this as resolved, since this is only ever called on the `requires`, not the `operation` as I had hypothesized. A comment to articulate this would be great!